From 2b94dc5ee5e79cd6d3c2adc696bb8bb4a9b334cf Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Sat, 25 Apr 2020 07:53:16 +0300
Subject: [PATCH] [SW-0] gaudi: asic: add hl_parse_uart()

Handle case where one tries to press [Esc] character
Then will print the module index
Use by BMC mapping of TTY to OAM

Change-Id: I6256cac4dafc75e322ef7f5c947e176603cbcb7a
---

diff --git a/gaudi/asic.c b/gaudi/asic.c
index a873e74..9a28adb 100644
--- a/gaudi/asic.c
+++ b/gaudi/asic.c
@@ -42,3 +42,25 @@
 
 	return factor > 2 ? factor : 2;
 }
+
+#define GAUDI_MEZZ_ID_MASK	0x7 /* We use only 3 bits out of 5 */
+#define GAUDI_MEZZ_ID_SHIFT	22
+#define ESC_KEY			0x1b
+void hl_parse_uart(int val)
+{
+	u8 mezz_id;
+	u32 strap;
+
+	strap = regmap_readl(mmPSOC_GLOBAL_CONF_BOOT_STRAP_PINS);
+	mezz_id = (strap >> GAUDI_MEZZ_ID_SHIFT) & GAUDI_MEZZ_ID_MASK;
+
+	switch (val) {
+	case ESC_KEY:
+		preboot_puts_raw("OAM[");
+		preboot_puth_raw(mezz_id, false);
+		preboot_puts("] Gaudi!\n");
+		break;
+	default:
+		break;
+	}
+}
